### PR TITLE
Insert app icons into Topics Screen #142

### PR DESCRIPTION
**Pull request by alouanemed**
Bounty #142 
Insert the designed icons into the Topics Screen, coded on bounty #108.
The screen will start with the grayed icons, which will then turn to color when selected.

final result : 
image

### CODE REVIEW

**Comment by fernandobuzi** [comment]
The fast @AlouaneMed. :) 

I think that when we do the polish at the end @intrications, this background should be made white.  Along with the login and sign up I'd say.

**Comment by alouanemed** [comment]
LOL :D 
@fernandobuzi  you mean change the background color for all the activities ?

**Comment by fernandobuzi** [comment]
No, not at all. Some of them, and only if possible. It's not a must, but these would look better on a white background.

**Comment by alouanemed** [comment]
@fernandobuzi 
like this ?
image

**Comment by fernandobuzi** [comment]
Something like that, but please don't change it now. We should leave it gray for now and merge this bounty as is, after it is reviewed.

File: app/src/main/java/com/kanshu/kanshu/TopicsActivity.java [hunk 1]
```diff
@@ -53,27 +53,40 @@ protected void onCreate(Bundle savedInstanceState) {
...
+                        mAdapter.notifyDataSetChanged();
                     } else {
                         //we are good let's add new one
                         mChosenTopicsListIDs.add(position);
-                        mTopicPic.setBorderColor(Color.parseColor("#D5FF79"));
-                        mTopicPic.setBorderWidth(8);
+
+                        String current_path = mTopicsList.get(position).getImgURL();
+                        String helper_str  =current_path.substring(0,current_path.length()-9);
+                        System.out.println("helper_str>" + helper_str);
```
Comment (intrications) [inline app/src/main/java/com/kanshu/kanshu/TopicsActivity.java]: Can you remove these log statements if they are no longer required?

File: app/src/main/res/layout/activity_signup.xml [hunk 2]
```diff
@@ -1,14 +1,14 @@
 <FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (intrications) [inline app/src/main/res/layout/activity_signup.xml]: Please revert the changes to activity_signup.xml. They look like they are just formatting changes.

**Comment by intrications** [comment]
When I first click on a topic it disappears before reappearing fully coloured.

Maybe you could change the images to greyscale in code rather than duplicating all the images:
http://stackoverflow.com/a/3391061/84021

It would reduce the size of the app significantly and probably avoid the problem I just mentioned.

**Comment by alouanemed** [comment]
@intrications where did you test the app ? 
because I think it's only an issue in the emulator 
no ?

**Comment by fernandobuzi** [comment]
@intrications @AlouaneMed I like the idea of getting rid of the grey icons and use code instead. Especially if it contributes to reduce the app's size.

**Comment by alouanemed** [comment]
Since We aren't going to need the gray icons I am going to delete them

**Comment by alouanemed** [comment]
Well , I think that's all :)

**Comment by intrications** [comment]
@alouanemed Probably doesn't matter anymore but I was testing on a Nexus 6.

**Comment by alouanemed** [comment]
@intrications Anyway, the gray method is better :+1:

**Comment by intrications** [comment]
@alouanemed I'm still getting the flash before the coloured image appears. Are you seeing that in the emulator?

**Comment by alouanemed** [comment]
@intrications Yeah I noticed that ,I think it's related to the bitmap cashing .It will appears only on the first time ,after if you click on the same icon it will appears without the flash !

**Comment by fernandobuzi** [comment]
Do you mean that on the simulator or the actual app @AlouaneMed. Because on the app that would be a problem, since people would choose each topic only once.

**Comment by alouanemed** [comment]
@fernandobuzi  both !
@intrications I think it's something related to picasso?

**Comment by intrications** [comment]
Yes. It is presumably when Picasso first loads the image from drawables
folder. Seems slower than it should be though even for that. When the
Activity first starts all the images are there immediately and they are
being displayed with Picasso as well, I assume.

**Comment by alouanemed** [comment]
@intrications  Yeah  , I guess so . Let me do some tests about that.

**Comment by alouanemed** [comment]
@intrications Well , I guess we can't load the generated bitmap into imageView using Picasso !

**Comment by intrications** [comment]
@alouanemed The delay was happening even when you were using pre-generated with images. So the conversion to greyscale isn't the problem.

Are you just using one ImageView and changing the image shown? How about using two ImageViews and layer them on top of each other and changing the visibility of them to only display the appropriate one? Not the most efficient memory usage but it should work.

**Comment by alouanemed** [comment]
@intrications let me test that

**Comment by intrications** [comment]
@alouanemed 

Also, are you sure you are using Picasso to set the greyscale image:

```
Bitmap mTopic_bmp = mTopicsController.toGrayscale(mTopicsList.get(position).getImgURL());
holder.IvTopicImg.setImageBitmap(mTopic_bmp);
```

That second line should use Picasso. That might be causing the delay.

**Comment by alouanemed** [comment]
@intrications  no I am not using Picasso to set the greyscale pic, I am using it only to set the drawable

**Comment by alouanemed** [comment]
@intrications  check this I think Picasso is causing the problem

**Comment by intrications** [comment]
Yep, that's fine. I'm not really surprised Picasso caused this. It's mainly good for downloading images from the web.

**Comment by alouanemed** [comment]
@intrications  True !
